pennylane/transforms/qcut.py

```diff
@@ -505,3 +505,162 @@ def contract_tensors(
     kwargs = {} if use_opt_einsum else {"like": tensors[0]}
 
     return contract(eqn, *tensors, **kwargs)
+
+
+CHANGE_OF_BASIS = qml.math.array(
+    [[1.0, 1.0, 0.0, 0.0], [-1.0, -1.0, 2.0, 0.0], [-1.0, -1.0, 0.0, 2.0], [1.0, -1.0, 0.0, 0.0]]
+)
+
+
+def _process_tensor(results, n_prep: int, n_meas: int):
+    """Convert a flat slice of execution results into a tensor.
+
+    This function performs the following steps:
+
+    1. Reshapes ``results`` into the shape ``(4,) * n_prep + (4**n_meas,)``
```

Comment:
Minor comment but suppose n_prep = 3 and n_meas = 2. Then the shape should be ((4,), (4,), (4,), (16,)) if I understand the function correctly. Is it standard to use the symbols * and + to denote repeating dimensions like this? Initially I had some trouble interpreting the + especially.

Reply:
Hey @angusjlowe, I had a very similar question which I think has since been hidden from this conversation but you might find useful: 

https://github.com/PennyLaneAI/pennylane/pull/2192/files#r809528123

Reply:
Ah this is just how Python tuples work -- so this would be (4,4,4,16). Nevermind!